solr/solrj/src/java/org/apache/solr/common/util/SimpleOrderedMap.java

```diff
@@ -68,24 +74,153 @@ public SimpleOrderedMap(Map.Entry<String, T>[] nameValuePairs) {
   public SimpleOrderedMap<T> clone() {
     ArrayList<Object> newList = new ArrayList<>(nvPairs.size());
     newList.addAll(nvPairs);
-    return new SimpleOrderedMap<>(newList);
+    return new SimpleOrderedMap<T>(newList);
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return nvPairs.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(final Object key) {
+    return this.indexOf((String) key) >= 0;
   }
 
   /**
-   * Returns a shared, empty, and immutable instance of SimpleOrderedMap.
+   * Returns {@code true} if this map maps one or more keys to the specified value.
    *
-   * @return Empty SimpleOrderedMap (immutable)
+   * @param value value whose presence in this map is to be tested
+   * @return {@code true} if this map maps one or more keys to the specified value
    */
-  public static SimpleOrderedMap<Object> of() {
-    return EMPTY;
+  @Override
+  public boolean containsValue(final Object value) {
+    int sz = size();
```

Comment:
instead of all these lines, could be a one-liner values().contains(value).  It'll be rare if ever for any Solr code to call this.